launcher/net/PasteUpload.cpp

```diff
@@ -8,44 +8,34 @@
 #include <QJsonDocument>
 #include <QFile>
 
-PasteUpload::PasteUpload(QWidget *window, QString text, QString key) : m_window(window)
+PasteUpload::PasteUpload(QWidget *window, QString text, QString url) : m_window(window), m_uploadUrl(url), m_text(text.toUtf8())
 {
-    m_key = key;
-    QByteArray temp;
-    QJsonObject topLevelObj;
-    QJsonObject sectionObject;
-    sectionObject.insert("contents", text);
-    QJsonArray sectionArray;
-    sectionArray.append(sectionObject);
-    topLevelObj.insert("description", "Log Upload");
-    topLevelObj.insert("sections", sectionArray);
-    QJsonDocument docOut;
-    docOut.setObject(topLevelObj);
-    m_jsonContent = docOut.toJson();
 }
 
 PasteUpload::~PasteUpload()
 {
 }
 
-bool PasteUpload::validateText()
-{
-    return m_jsonContent.size() <= maxSize();
-}
-
 void PasteUpload::executeTask()
 {
-    QNetworkRequest request(QUrl("https://api.paste.ee/v1/pastes"));
+    QNetworkRequest request{QUrl(m_uploadUrl)};
     request.setHeader(QNetworkRequest::UserAgentHeader, BuildConfig.USER_AGENT_UNCACHED);
 
-    request.setRawHeader("Content-Type", "application/json");
-    request.setRawHeader("Content-Length", QByteArray::number(m_jsonContent.size()));
-    request.setRawHeader("X-Auth-Token", m_key.toStdString().c_str());
+    QHttpMultiPart *multiPart = new QHttpMultiPart{QHttpMultiPart::FormDataType};
```

Comment:
pacebin literally just uses the POST body itself, will need to test if it works properly here (not familiar with multipart so maybe it does)

Reply:
> pacebin literally just uses the POST body itself, will need to test if it works properly here (not familiar with multipart so maybe it does)

If you really want to be able to support both 0x0 and pacebin then you'll either have to change pacebin to support getting the file from the file parameter, or make it configurable 

You'll need to change pacebin to support posting to the file parameter then, or change the feature to make the API configurable.

Reply:
I will probably change pacebin